code/controllers/subsystems/skybox.dm

```diff
@@ -0,0 +1,82 @@
+
+//Exists to handle a few global variables that change enough to justify this. Technically a parallax, but it exhibits a skybox effect.
+SUBSYSTEM_DEF(skybox)
+	name = "Space skybox"
+	init_order = INIT_ORDER_SKYBOX
+	flags = SS_NO_FIRE
+	var/list/skybox_cache = list()
+
+/datum/controller/subsystem/skybox/Initialize()
+	. = ..()
+
+/datum/controller/subsystem/skybox/Recover()
+	skybox_cache = SSskybox.skybox_cache
+
+/datum/controller/subsystem/skybox/proc/get_skybox(z)
+	if(!skybox_cache["[z]"])
+		skybox_cache["[z]"] = generate_skybox(z)
+		if(global.using_map.use_overmap)
+			var/obj/effect/overmap/visitable/O = map_sectors["[z]"]
+			if(istype(O))
+				for(var/zlevel in O.map_z)
+					skybox_cache["[zlevel]"] = skybox_cache["[z]"]
+	return skybox_cache["[z]"]
+
+/datum/controller/subsystem/skybox/proc/generate_skybox(z)
+	var/datum/skybox_settings/settings = global.using_map.get_skybox_datum(z)
```

Comment:
Yes, I wrote that part. Z-level datums can produce custom skybox images and such, and so can using_map datums. 

The easy way to do the 'object in the skybox' would be to have the planet and spacestation as two separate sectors in the same turf (might want to sprite their overmap/visitable objects so that looks nice) and they don't just overlap 100%. Whenever overmap objects are in the same turf on the overmap, the skybox asks them if they have an image they want to use to represent themselves in the skybox of each other's z-levels, and draws that into the skybox.

Bay actually has a bunch of uncolored planets plus rings, ground features, etc, because they have a random planet generator. Might want to swipe those and color them for Sif.

![dreamseeker_2020-03-24_22-14-25](https://user-images.githubusercontent.com/15028025/77763056-e01ce880-7010-11ea-9ad0-795c4acd0410.png)

An example ship in the skybox, because the ship is in the same overmap turf as the station I'm on, during an electrical storm.